app/packs/src/apps/mydb/collections/CollectionSubtree.js

```diff
@@ -205,48 +162,51 @@ export default class CollectionSubtree extends React.Component {
     CollectionActions.updateCollectrionTree(newIds)
   }
 
-  synchronizedIcon() {
-    let sharedUsers = this.state.root.sync_collections_users
-    return (
-      sharedUsers && sharedUsers.length > 0
-        ? <OverlayTrigger placement="bottom" overlay={UserInfos({ users: sharedUsers })}>
-          <i className="fa fa-share-alt" style={{ float: "right" }}></i>
-        </OverlayTrigger>
-        : null
-    )
-  }
-
-
   render() {
-    const { fakeRoot } = this.props;
-    const { label, root } = this.state;
-    let { visible } = this.state;
+    const { label, root, visible, selected } = this.state;
+    const sharedUsers = root.sync_collections_users;
+    const children = this.children();
+
+    const showGatePushButton = root && root.is_locked && label === 'chemotion-repository.net';
 
-    let style;
-    if (!visible) {
-      style = {
-        display: 'none',
-        marginBottom: 0
-      };
-    }
-    const gated = root && root.is_locked && label == 'chemotion-repository.net' ?
-      <GatePushBtn collection_id={root.id} /> : null;
     return (
       <div className="tree-view" key={root.id}>
-        {this.takeOwnershipButton()}
-
-        <div id={`tree-id-${root.label}`} className={"title " + this.selectedCssClass()}
-          onClick={this.handleClick}>
-          {this.expandButton()}
-          {this.synchronizedIcon()}
-          {gated}
-          {label}
+        <div
+          id={`tree-id-${root.label}`}
+          className={`title ${selected ? 'selected' : ''} d-flex align-items-baseline`}
+          onClick={this.handleClick}
+        >
+          {showGatePushButton && (<GatePushBtn collection_id={root.id} />)}
```

Comment:
I am not too fond of this `{conditional && (renderSomething)}` pattern.
My preferred solution is creating a function
```javascript
const gatePushButton = () => {
  showButton = some_conditional
  if (!showButton) return null;

  return (<GatePushBtn ...>)
}
```

This makes the render method way easier to read.

What's your opinion? (and maybe the whole team's opinion)

Reply:
I have very strong feelings against having a four lines of code in a remote location somewhere in the class. What I found in a lot was that people added additional functionality and that tiny function driftet further and further away from the render method obscuring what the contained DOM actually consists of. Having the complete DOM/component tree within the render method is a nice separation of concern, this is where devs look if they want to change the markup. If you have a complex decision tree that determines what to render case a, b c,..., than yes it is likely bloating the render function. But if it's a simple condition then the one liner is way more readable to me.
The whole return null pattern is especially cumbersome to me, it emphasizes what not to do. I'm no react veteran and maybe my preferences are influenced by vue.js standards, but I'm pretty sure I haven't seen it anywhere outside of chemotion.

Reply:
We have already gone to great lengths to eliminate this kind of code elsewhere reducing the overall size of classes significantly.